src/aind_behavior_experiment_launcher/behavior_launcher/slims_picker.py

```diff
@@ -0,0 +1,270 @@
+            password (Optional[str]): slims password. Default sto SLIMS_PASSWORD environment variable if not provided
+            **kwargs: Additional keyword arguments.
+        """
+
+        super().__init__(launcher, ui_helper=ui_helper, **kwargs)
+
+        self.slims_client = self.connect_to_slims(slims_url, username, password)
+
+        # initialize properties
+        self._slims_mouse = None
+        self._slims_session = None
+
+    def connect_to_slims(self, url: str = None, username: str = None, password: str = None) -> SlimsClient:
+        """
+        Connect to Slims with optional username and password or use environment variables
+
+        Args:
+            url(Optional[str]): slims url. Defaults to dev version of slims if not provided
+            username (Optional[str]): slims username. Defaults to SLIMS_USERNAME environment variable if not provided
+            password (Optional[str]): slims password. Defaults to SLIMS_PASSWORD environment variable if not provided
+
+        Returns:
+            SlimsClient: slims client instance.
+
+        Raises:
+            Exception: If invalid credentials or error reading from slims
+        """
+
+        try:
+            logger.info("Attempting to connect to Slims")
+            slims_client = SlimsClient(
+                url=url,
+                username=username if username else os.environ["SLIMS_USERNAME"],
+                password=password if password else os.environ["SLIMS_PASSWORD"],
+            )
+            slims_client.fetch_model(models.SlimsMouseContent, barcode="00000000")
+
+        except exceptions.SlimsAPIException as e:
+            if "Status 401 – Unauthorized" in str(e):  # catch error if username and password are incorrect
+                raise Exception(
+                    f"Exception trying to read from Slims: {e}.\n"
+                    f" Please check credentials:\n"
+                    f"Username: {username if username else os.environ['SLIMS_USERNAME']}\n"
+                    f"Password: {password if password else os.environ['SLIMS_PASSWORD']}"
+                )
+            else:
+                raise Exception(f"Exception trying to read from Slims: {e}.\n")
+
+        logger.info("Successfully connected to Slims")
+
+        return slims_client
+
+    @property
+    def slims_mouse(self) -> models.mouse:
+        """
+        Returns slims mouse model being used to load session
+
+        Returns:
+            models.Mouse: slims mouse model object
+        """
+
+        return self._slims_mouse
+
+    @property
+    def slims_session(self) -> models.behavior_session.SlimsBehaviorSession:
+        """
+        Returns slims session model being used to load task logic
+
+        Returns:
+            models.behavior_session.SlimsBehaviorSession: slims session model object
+        """
+
+        return self._slims_session
+
+    def pick_rig(self) -> _R:
```

Comment:
This function should be changed to:
1. prompt the rig id from user (See the todo below tho)
2. Find the rig in slims (raises SlimsRecordNotFound)
3. From the rig in slims, fetch the latest attachment (not sure what is returned here. If empty throw a ValueError)
4. Attempt to deserialize the attachment to the launcher.rig_schema_model (Throws pydantic validation error)
5. If it fails, perhaps allow people to pick a config from the last N=10 (?)